webapp/javascript/services/render.ts

```diff
@@ -39,7 +45,7 @@ export async function renderSingle(
     z.object({ timeline: TimelineSchema })
   )
     .merge(z.object({ telemetry: z.object({}).passthrough().optional() }))
-    .merge(z.object({ groups: z.object({}).passthrough().optional() }))
+    .merge(z.object({ groups: GroupsSchema.optional() }))
```

Comment:
my understanding is that this case doesn't use `groups` data at all, since it's deal with by renderExplore, so do we need this?

Reply:
correct, but i think its nice to have it here, because `groups` are part of the response, even though we don't use it 
https://github.com/pyroscope-io/pyroscope/pull/1244#discussion_r919350675

Reply:
I disagree exactly for those reasons, if it's not used, then it's just adding complexity for no reason. Everything that is in the code base must be there for a reason.

A prime example of this is golang and its clean interfaces (eg  [io.Writer](https://pkg.go.dev/io#Writer)), by reducing the API scope you plug whatever you want, as long as it fulfills the interface.